lib/declare_schema/model/habtm_model_shim.rb

```diff
@@ -4,22 +4,25 @@ module DeclareSchema
   module Model
     class HabtmModelShim
       class << self
-        def from_reflection(refl)
-          new(refl.join_table, [refl.foreign_key, refl.association_foreign_key],
-                               [refl.active_record.table_name, refl.class_name.constantize.table_name])
+        def from_reflection(reflection, connection:)
+          new(reflection.join_table,
+              [reflection.foreign_key, reflection.association_foreign_key],
+              [reflection.active_record.table_name, reflection.class_name.constantize.table_name],
+              connection: connection)
         end
       end
 
-      attr_reader :join_table, :foreign_keys, :parent_table_names
+      attr_reader :join_table, :foreign_keys, :parent_table_names, :connection
 
-      def initialize(join_table, foreign_keys, parent_table_names)
+      def initialize(join_table, foreign_keys, parent_table_names, connection:)
```

Comment:
This feels like an internal change, but technically might be "breaking" as it now adds a new required argument to this class.

Thoughts?

Reply:
This is an internal class only (not documented in the public interface) so I think it's fine.

Reply:
This is an internal-only class... but it turns out that was not needed any more, so I've dropped it. (`HabtmModelShim` needs to be polymorphic with `Model`, and when I added this, there was a `connection` method on `Model`. That's gone now.)